src/components/fields/SchemaField.js

```diff
@@ -128,38 +129,50 @@ DefaultTemplate.defaultProps = {
   displayLabel: true,
 };
 
-function SchemaField(props) {
-  const {uiSchema, errorSchema, idSchema, name, required, registry} = props;
-  const {definitions, fields, formContext, FieldTemplate = DefaultTemplate} = registry;
-  const schema = retrieveSchema(props.schema, definitions);
-  const FieldComponent = getFieldComponent(schema, uiSchema, fields);
-  const {DescriptionField} = fields;
-  const disabled = Boolean(props.disabled || uiSchema["ui:disabled"]);
-  const readonly = Boolean(props.readonly || uiSchema["ui:readonly"]);
-  const autofocus = Boolean(props.autofocus || uiSchema["ui:autofocus"]);
-
-  if (Object.keys(schema).length === 0) {
-    // See #312: Ensure compatibility with old versions of React.
-    return <div/>;
+class SchemaField extends Component {
+  shouldComponentUpdate(nextProps, nextState) {
+    let _props = {...this.props};
+    let _nextProps = {...nextProps};
+    // if schema and formData are the equal
+    // idSchema and errorSchema will be equal
+    delete _props.idSchema;
+    delete _nextProps.idSchema;
+    return !deepEquals(_props, _nextProps);
```

Comment:
The comment suggests that "schema" and "formData" didn't change => "idSchema" and "errorSchema" didn't change. So, comparing "schema" and "formData" is enough to tell wheter "idSchema" and "errorSchema" have change or not.
But, not comparing errorSchema will fail some test. I haven't figure out the reason yet. And, the performance with comparing errorSchema looks acceptable. 
I will update the comment.